packages/kbn-alerts-ui-shared/src/common/hooks/use_find_alerts_query.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useQuery } from '@tanstack/react-query';
+import type { HttpStart } from '@kbn/core-http-browser';
+import type { ToastsStart } from '@kbn/core-notifications-browser';
+import { ISearchRequestParams } from '@kbn/search-types';
+import { SearchResponseBody } from '@elastic/elasticsearch/lib/api/types';
+import { BASE_RAC_ALERTS_API_PATH } from '../constants';
+
+export interface UseFindAlertsQueryProps {
+  http: HttpStart;
+  toasts: ToastsStart;
+  enabled?: boolean;
+  params: ISearchRequestParams & { feature_ids?: string[] };
+}
+
+/**
+ * A generic hook to find alerts
+ *
+ * Still applies alerts authorization rules but, unlike triggers_actions_ui's `useFetchAlerts` hook,
+ * allows to perform arbitrary queries
+ */
+export const useFindAlertsQuery = <T>({
```

Comment:
do we have concrete types for this? or it has to be a generic?

Reply:
Here the generic is used to type the response depending on the aggregations you choose to fetch from the endpoint 🙂